pkg/set_test.go

```diff
@@ -0,0 +1,164 @@
+package adt
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+)
+
+func ExampleNewSet() {
+	s1 := NewSet(1, 2, 3)
+	fmt.Println(reflect.TypeOf(s1))
+
+	s2 := NewSet("a", "b", "c")
+	fmt.Println(reflect.TypeOf(s2))
+
+	s3 := NewSet[rune]() // example of initialization with 0 elements
+	fmt.Println(reflect.TypeOf(s3))
+
+	// Output:
+	// adt.Set[int]
+	// adt.Set[string]
+	// adt.Set[int32]
+}
+
+func ExampleAdd() {
```

Comment:
You should name example functions as `ExampleSet_Add`, `ExampleSet_Remove`, `ExampleSet_Clone`,...
That way Go docs will display these examples next to those methods on the page.

Reply:
fixed.